tests/src/Services/AuthenticationServiceTest.php

```diff
@@ -199,45 +174,63 @@ public function testItReturnsAnUser(): void
             ->willReturn($this->userEntityMock);
         $this->userRepositoryMock->expects($this->once())->method('update')->with($this->userEntityMock);
 
-        $this->sessionServiceMock->expects($this->once())->method('getCurrentSession')->willReturn($this->sessionMock);
-        $this->sessionServiceMock->expects($this->once())->method('setIsCookieBasedAuthn')->with(false);
-        $this->sessionServiceMock->expects($this->once())->method('setIsAuthnPerformedInPreviousRequest')->with(true);
-
         $this->claimTranslatorExtractorMock->expects($this->once())->method('extract')
             ->with(['openid'], $this->isType('array'))
             ->willReturn([]);
 
-        $this->authProcServiceMock->method('processState')->willReturn(self::STATE);
-
         $this->assertSame(
-            $this->prepareMockedInstance()->getAuthenticateUser($this->serverRequestMock),
+            $this->prepareMockedInstance()->getAuthenticateUser(self::STATE),
             $this->userEntityMock,
         );
     }
 
     /**
-     * @throws \SimpleSAML\Error\AuthSource
+     * @return void
+     * @throws Exception
+     * @throws \JsonException
      * @throws \SimpleSAML\Error\BadRequest
      * @throws \SimpleSAML\Error\NotFound
+     * @throws \SimpleSAML\Module\oidc\Server\Exceptions\OidcServerException
      */
     public function testItThrowsIfClaimsNotExist(): void
     {
-        $this->authSourceMock->method('getAuthId')->willReturn('theAuthId');
-
-        $this->authSimpleMock->expects($this->once())->method('isAuthenticated')->willReturn(false);
-        $this->authSimpleMock->expects($this->once())->method('login');
-        $this->authSimpleMock->expects($this->once())->method('getAuthSource')->willReturn($this->authSourceMock);
-
-        $this->sessionServiceMock->expects($this->once())->method('setIsCookieBasedAuthn')->with(false);
-        $this->sessionServiceMock->expects($this->once())->method('setIsAuthnPerformedInPreviousRequest')->with(true);
-
         $invalidState = self::STATE;
         unset($invalidState['Attributes'][self::USER_ID_ATTR]);
 
-        $this->authProcServiceMock->expects($this->once())->method('processState')->willReturn($invalidState);
-
         $this->expectException(Exception::class);
 
-        $this->prepareMockedInstance()->getAuthenticateUser($this->serverRequestMock);
+        $this->prepareMockedInstance()->getAuthenticateUser($invalidState);
+    }
+
+    /**
+     * @return void
+     * @throws \JsonException
+     * @throws \SimpleSAML\Error\BadRequest
+     * @throws \SimpleSAML\Error\NotFound
+     * @throws \SimpleSAML\Module\oidc\Server\Exceptions\OidcServerException
+     */
+    public function testItAuthenticates(): void
+    {
+        $authenticationServiceMock = $this->getMockBuilder(AuthenticationService::class)
```

Comment:
I don't follow why you are mocking the class under tests.  I think you can use use the an instance of AuthenticationService with mocked dependencies, which is what `$this->prepareMockedInstance()` should give you.

Reply:
the `$this->prepareMockedInstance()` returns the actual object. What I needed for this test was a mocked object so that  I could use the `expects` function